plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,313 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import io.airlift.log.Logger;
+import io.trino.Session;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.io.MoreFiles.deleteDirectoryContents;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.deltalake.DeltaLakeConnectorFactory.CONNECTOR_NAME;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogJsonEntryPath;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class BaseDeltaLakeRegisterTableProcedureTest
+        extends AbstractTestQueryFramework
+{
+    private static final Logger LOG = Logger.get(BaseDeltaLakeRegisterTableProcedureTest.class);
+
+    protected static final String CATALOG_NAME = "test_delta";
+    protected static final String SCHEMA = "register_table_" + randomTableSuffix();
+
+    protected File metastoreDir;
+    protected HiveMetastore metastore;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(SCHEMA)
+                .build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
+
+        this.metastoreDir = new File(queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake_data").toString());
+        this.metastore = createTestMetastore();
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin());
+
+        queryRunner.createCatalog(CATALOG_NAME, CONNECTOR_NAME, getConnectionProperties());
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+
+        return queryRunner;
+    }
+
+    protected abstract HiveMetastore createTestMetastore();
```

Comment:
Put it after `getConnectorProperties` and document that it needs to return a metastore client talking to the same metastore as the one configured via `getConnectorProperties`.

Reply:
Used this approach https://github.com/trinodb/trino/pull/14779#discussion_r1021391249 and modified the method signature.